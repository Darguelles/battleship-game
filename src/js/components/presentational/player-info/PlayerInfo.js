import React, {Component} from "react";
import {Card, ProgressBar} from 'react-materialize';
import PropTypes from 'prop-types';
import './PlayerInfo.scss'

class PlayerInfo extends Component {

    constructor(props) {
        super(props);
    }

    render() {
        let loader = null;
        if (this.props.attempts === 'INFINITE') {
            loader = <ProgressBar progress={100}/>
        } else {
            loader = <ProgressBar progress={parseInt(this.props.attempts)}/>
        }

        return (
            <Card className='teal lighten-5 black-text' title={this.props.playerName}>
                <div className={"player-info"}>
                    <h6>Turns</h6>
                    <label>{this.props.attempts}</label>
                    {loader}
                    <h6>Hits</h6>
                    <label className={"hit"}>{this.props.hits}</label>
                    <h6>Failures</h6>
                    <label className={"failure"}>{this.props.failures}</label>
                    <h6>Start time</h6>
                    <label>{this.props.startTime}</label>
                    <h6>Emd time</h6>
                    <label>{this.props.endTime}</label>
                </div>
            </Card>
        );
    }
}

PlayerInfo.propTypes = {
    playerName: PropTypes.string,
    attempts: PropTypes.any,
    hits: PropTypes.number,
    failures: PropTypes.number,
    startTime: PropTypes.string,
    endTime: PropTypes.string,
};

export default PlayerInfo;