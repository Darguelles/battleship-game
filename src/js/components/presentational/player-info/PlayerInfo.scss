.player-info {
    label {
        font-size: 20px;
    }
    .hit {
        color: darkgreen;
    }
    .failure {
        color: darkred;
    }
}