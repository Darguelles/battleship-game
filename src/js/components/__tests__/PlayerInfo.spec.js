import {configure, shallow} from 'enzyme';
import React from 'react';
import PlayerInfo from '../presentational/player-info/PlayerInfo'
import Adapter from 'enzyme-adapter-react-16';
import Helpers from "../container/helpers";
import renderer from 'react-test-renderer';


configure({ adapter: new Adapter() });

describe('PlayerInfo > presentation behavior', () => {

    it('should renders correctly', () => {
        const rendered = renderer.create(
            <PlayerInfo/>
        );

        expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('should show startDate in correct format', () => {
        const formattedDate = Helpers.getCurrentDate();
        const regX = new RegExp(/(\d+)(-|\/)(\d+)(?:-|\/)(?:(\d+)\s+(\d+):(\d+)(?::(\d+))?(?:\.(\d+))?)?/);
        expect(formattedDate).toMatch(regX);
    });

});
