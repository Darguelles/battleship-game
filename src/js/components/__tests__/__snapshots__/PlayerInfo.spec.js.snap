// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerInfo > presentation behavior should renders correctly 1`] = `
<div
  className="teal lighten-5 black-text card"
  id="card-player-info"
>
  <div
    className="card-content"
  >
    <div>
      <div
        className="player-info"
      >
        <h6>
          Turns
        </h6>
        <label
          id="lbl-attempts"
        />
        <div
          className="progress lighten-5"
        >
          <div
            className=""
            style={undefined}
          />
        </div>
        <h6>
          Hits
        </h6>
        <label
          className="hit"
          id="lbl-hits"
        />
        <h6>
          Failures
        </h6>
        <label
          className="failure"
          id="lbl-failures"
        />
        <h6>
          Start time
        </h6>
        <label
          id="lbl-starttime"
        />
        <h6>
          Emd time
        </h6>
        <label
          id="lbl-endtime"
        />
      </div>
    </div>
  </div>
  <div
    className="card-reveal"
  >
    <span
      className="card-title grey-text text-darken-4"
    >
      <i
        className="material-icons right"
      >
        close
      </i>
    </span>
  </div>
</div>
`;
